lib/Ast_to_Mir.ml

```diff
@@ -3,11 +3,36 @@
 open Core_kernel
 open Mir
 
+let trans_op (op : Ast.operator) =
+  match op with
+  | Ast.Plus -> Plus
+  | Ast.Minus -> Minus
+  | Ast.Times -> Times
+  | Ast.Divide -> Divide
+  | Ast.Modulo -> Modulo
+  | Ast.Or -> Or
+  | Ast.And -> And
+  | Ast.Equals -> Equals
+  | Ast.NEquals -> NEquals
+  | Ast.Less -> Less
+  | Ast.Leq -> Leq
+  | Ast.Greater -> Greater
+  | Ast.Geq -> Geq
+  | _ ->
+      raise_s [%message "Should never try to translate " (op : Ast.operator)]
+
 let rec trans_expr {Ast.expr_typed; _} =
   match expr_typed with
   | Ast.TernaryIf (cond, ifb, elseb) ->
       TernaryIf (trans_expr cond, trans_expr ifb, trans_expr elseb)
-  | Ast.BinOp (lhs, op, rhs) -> BinOp (trans_expr lhs, op, trans_expr rhs)
+  | Ast.BinOp (lhs, op, rhs) -> (
+      let lhs, rhs = (trans_expr lhs, trans_expr rhs) in
+      match op with
+      | Ast.LDivide | Ast.EltTimes | Ast.EltDivide | Ast.Pow | Ast.Not
+       |Ast.Transpose ->
+          let fnname = Sexp.to_string_hum [%sexp (op : Ast.operator)] in
```

Comment:
Wouldn't you get capitalization in the function name this way? Is that the name we use in the Math library?

Reply:
Yeah, there will need to be a separate, Stan Math-specific translation step. We get to choose what they are in the MIR, so it seemed to make sense to follow the AST etc. We can lowercase them though, that might make more sense (some will maybe still need translation).

Reply:
on second thought I think I kind of like that they are upper cased as you can tell that they're a bit different from your typical math functions. Leaving for now...